packages/edit-post/src/components/header/contextual-toolbar/index.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import classnames from 'classnames';
+
+/**
+ * WordPress dependencies
+ */
+import {
+	BlockToolbar,
+	privateApis as blockEditorPrivateApis,
+} from '@wordpress/block-editor';
+import { useEffect, useRef } from '@wordpress/element';
+import { Button, Popover } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { next, previous } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import { unlock } from '../../../lock-unlock';
+
+const { useHasBlockToolbar } = unlock( blockEditorPrivateApis );
+
+function ContextualToolbar( {
+	blockSelectionStart,
```

Comment:
This prop is only used here and not in the outer component, it would have been better to move the useSelect call here

Reply:
I considered it, but thought combining the useSelects might be better performance-wise.

Reply:
It's a tradeoff but I believe unless we have measurable performance impact, it's better to favor collocating the state.

Reply:
Also this selector will not be called at all if the component is not rendered but it will always be called if it's done in the upper component.